sdk/storage/azure-storage-queues/cgmanifest.json

```diff
@@ -5,33 +5,21 @@
       "Component": {
         "Type": "git",
         "git": {
-          "RepositoryUrl": "https://github.com/cpplint/cpplint",
-          "CommitHash": "8456631ba3a12842da08238362a90d33fcf47062"
+          "RepositoryUrl": "https://github.com/openssl/openssl",
+          "CommitHash": "d82e959e621a3d597f1e0d50ff8c2d8b96915fd7"
         }
       },
-      "DevelopmentDependency": true
+      "DevelopmentDependency": false
     },
     {
       "Component": {
-        "Type": "other",
-        "Other": {
-          "Name": "clang-format",
-          "Version": "9.0.0-2",
-          "DownloadUrl": "https://ubuntu.pkgs.org/18.04/ubuntu-updates-universe-amd64/clang-format-9_9-2~ubuntu18.04.2_amd64.deb.html"
-        }
-      },
-      "DevelopmentDependency": true
-    },
-    {
-      "Component": {
-        "Type": "other",
-        "Other": {
-          "Name": "doxygen",
-          "Version": "1.8.20",
```

Comment:
We do use clang-format and doxygen as a development dependency. Consider keeping these two as is, and **only adding** the two non-Dev dependencies (libxml2 and openssl).

Reply:
We already have clang-format and doxygen in `azure-core/cgmanifest.json`. Storage SDK depends on azure-core.
Do we still need to add these items in every package's cgmanifest file?

Reply:
That's a good open question and it ties into whether we need multiple or a single notice.txt file.
Imo, while we figure out the conclusion on this, let's keep all the external dependencies that each package needs (at least ones that already exist), and not remove any yet.

Reply:
According to the [component governance instructions](https://docs.opensource.microsoft.com/tools/cg/component-detection/cgmanifest/), the cgmanifest.json file should be used to list undiscoverable components, which currently includes packages referenced by vcpkg.

Based on my reading, the cgmanifest.json file should NOT include components which the component governance plugin can automatically discover.

Looking at the output of a [component governance run](https://dev.azure.com/azure-sdk/public/_build/results?buildId=3247290&view=logs&j=35d7e651-8a8e-56d5-4b64-d16694245812&t=925a2285-307c-5495-84e4-d4476e273199), it appears that the following packages are auto-discovered:

- cspell
- websockets (from requirements.txt in the core test repo).

The following packages are from cgmanifest.json files:

- googletest, referenced in:
  - azure-core-amqp
  - azure-security-keyvault-certificates
  - azure-security-keyvault-administration
  - azure-template
  - azure-security-keyvault-keys
  - azure-security-keyvault-secrets
  - azure-core-test
  - azure-core-tracing-opentelemetry
- doxygen, referenced in:
  - azure-security-keyvault-certificates
  - azure-core
  - azure-storage-files-shares
  - azure-storage-files-datalake
  - azure-security-keyvault-administration
  - template
  - azure-security-keyvault-keys
  - azure-storage-blobs
  - azure-storage-common
  - azure-security-keyvault-secrets
  - azure-storage-queues
- clang-format, referenced in
  - azure-security-keyvault-certificates/cgmanifest.json
  - azure-core/cgmanifest.json
  - azure-storage-files-shares/cgmanifest.json
  - azure-security-attestation/cgmanifest.json
  - azure-storage-files-datalake/cgmanifest.json
  - azure-security-keyvault-administration/cgmanifest.json
  - azure-template/cgmanifest.json
  - azure-security-keyvault-keys/cgmanifest.json
  - azure-storage-blobs/cgmanifest.json
  - azure-storage-common/cgmanifest.json
  - azure-security-keyvault-secrets/cgmanifest.json
  - azure-core-test/cgmanifest.json
  - azure-storage-queues/cgmanifest.json
- cpplint, referenced in:
  - azure-storage-files-shares/cgmanifest.json
  - azure-storage-files-datalake/cgmanifest.json
  - azure-storage-blobs/cgmanifest.json
  - azure-storage-common/cgmanifest.json
  - azure-storage-queues/cgmanifest.json
- openssl, referenced in:
  - azure-core/cgmanifest.json
  - azure-security-attestation/cgmanifest.json
  - azure-storage-common/cgmanifest.json
- argagg, referenced in:
  - core/perf
- open-telemetry-cpp, referenced in:
  - azure-core-tracing-opentelemetry
- macro-utils-c, azure-c-shared-utility, umock-c
  - azure-core-amqp
- wil, httpbin, gcovr, nlohmann, CLIUtils, googletest, curl, referenced in
  - azure-core
 
Attestation also has a reference to doxygen 1.8.18 which is listed separately.

The component governance scans are disabled, except in the "create apiview" workflow. I suspect that we can refine this a great deal by centralizing a lot of the components, especially the ones that are a part of our ES workflow (clang-format, doxygen, etc).

Reply:
I assume this doesn't block this PR. We can do other improvements in a separate PR.